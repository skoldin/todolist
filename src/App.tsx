import React, { useState, useEffect } from "react";
import Amplify, { API, graphqlOperation } from "aws-amplify";
import aws_config from "./aws-exports";
import * as mutations from "./graphql/mutations";
import * as queries from "./graphql/queries";
import List from "./components/List";
import AddItem from "./components/AddItem";
import "./App.css";
import "bootstrap/dist/css/bootstrap.css";

Amplify.configure(aws_config);

const App = () => {
  const defaultState: Array<object> = [];
  const [items, setListItems] = useState(defaultState);

  useEffect(() => {
    const data: any = API.graphql(graphqlOperation(queries.listTodos));

    data.then((data: { data: { listTodos: { items: object[] } } }) => {
      const items = data.data.listTodos.items;

      setListItems(items);
    });
  });

  const addItem: (item: Item) => void = item => {
    API.graphql(graphqlOperation(mutations.createTodo, { input: item }));
  };

  const updateItem: (item: Item) => void = item => {
    API.graphql(graphqlOperation(mutations.updateTodo, { input: item }));
  };

  const deleteItem: (id: string) => void = id => {
    const item = {
      id
    };

    API.graphql(graphqlOperation(mutations.deleteTodo, { input: item }));
  };

  return (
    <div className="container py-5">
      <h1>Things to do:</h1>
      <AddItem handleAdd={addItem} />
      <List listItems={items} updateItem={updateItem} deleteItem={deleteItem} />
    </div>
  );
};

export default App;
