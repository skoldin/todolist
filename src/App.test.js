import React from "react";
import App from "./App";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

let Component = renderer.create(<App />);
let wrapper;

beforeEach(() => {
  wrapper = mount(<App />);
});

it("render matches the snapshot", () => {
  expect(Component.toJSON()).toMatchSnapshot();
});
