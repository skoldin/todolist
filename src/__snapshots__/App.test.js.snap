// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render matches the snapshot 1`] = `
<div
  className="container py-5"
>
  <h1>
    Things to do:
  </h1>
  <div
    className="d-flex mb-3"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Add a thing to do"
      type="text"
      value=""
    />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</div>
`;
